settings/serverSettings.ts

```diff
@@ -1,94 +1,100 @@
 // This is where server-side only, potentially sensitive settings enter from the environment
 // DO NOT store sensitive strings in this file itself, as it is checked in to git!
 
-import fs from "fs-extra"
 import path from "path"
+import dotenv from "dotenv"
+dotenv.config()
 
-export const BASE_DIR = path.resolve(__dirname, "../..")
-const absoluteSettingsPath = path.resolve(BASE_DIR, "serverSettings.json")
-const localOverrides: any = fs.existsSync(absoluteSettingsPath)
-    ? fs.readJsonSync(absoluteSettingsPath)
-    : {}
+import * as clientSettings from "./clientSettings"
+import { parseIntOrUndefined } from "../clientUtils/Util"
 
-export const ADMIN_SERVER_PORT = localOverrides.ADMIN_SERVER_PORT ?? 3030
-export const ADMIN_SERVER_HOST = localOverrides.ADMIN_SERVER_HOST ?? "localhost"
-export const BAKED_BASE_URL =
-    localOverrides.BAKED_BASE_URL ??
-    `http://${ADMIN_SERVER_HOST}:${ADMIN_SERVER_PORT}`
+const serverSettings = process.env ?? {}
 
-export const BAKED_GRAPHER_URL =
-    localOverrides.BAKED_GRAPHER_URL ?? `${BAKED_BASE_URL}/grapher`
+export const BASE_DIR: string = path.resolve(__dirname, "../..")
+export const ENV: "development" | "production" = clientSettings.ENV
 
-export const ADMIN_BASE_URL =
-    localOverrides.ADMIN_BASE_URL ??
-    `http://${ADMIN_SERVER_HOST}:${ADMIN_SERVER_PORT}`
+export const ADMIN_SERVER_PORT: number = clientSettings.ADMIN_SERVER_PORT
+export const ADMIN_SERVER_HOST: string = clientSettings.ADMIN_SERVER_HOST
+export const BAKED_BASE_URL: string = clientSettings.BAKED_BASE_URL
```

Comment:
I went for double-exporting some settings (i.e. export them from both `clientSettings.ts` and `serverSettings.ts` under the same name), but am not sure if this is the right call.
Maybe/probably it's better to just export a setting like `ADMIN_SERVER_PORT` in `clientSettings` only, and then change all its imports over to `clientSettings`.
I would certainly like to hear which way you prefer.

But at least we no longer have to double-define these settings, since all settings are in a single `.env` file.

Reply:
Ah, the `ADMIN_SERVER_PORT` is interesting. I don't really have a preference. The way you've done it seems good to me – it's the same value no matter where you import it from. Importing `ADMIN_SERVER_PORT` from client settings is a bit weird, so I don't have a problem with double-exporting it as you've done it.